tls/s2n_quic_support.c

```diff
@@ -67,6 +67,17 @@ int s2n_connection_get_quic_transport_parameters(struct s2n_connection *conn,
     return S2N_SUCCESS;
 }
 
+int s2n_connection_set_secret_callback(struct s2n_connection *conn, s2n_secret_cb cb_func, void *ctx)
+{
+    notnull_check(conn);
+    notnull_check(cb_func);
+
+    conn->secret_cb = cb_func;
```

Comment:
Since this is only called when quic is enabled, would it be better to `ENSURE(conn->config->quic_enabled)` so callers can error immediately rather than silently ignore this callback?

That would also make it easier to call in the code: if there's a callback set, then we know it met all of the requirements to be able to be set so just call it (rather than check if quic_enabled again).

Implementing things like [`SSLKEYLOGFILE`](https://developer.mozilla.org/en-US/docs/Mozilla/Projects/NSS/Key_Log_Format) could be another potential future use for this function, right?

Reply:
It's only allowed when quic is enabled, but I worry about restricting the order that customers can call configuration APIs. If I require quic to be enabled when this is called, this would be invalid:
```
s2n_connection_set_secret_cb(conn, cb, context);
s2n_connection_set_config(conn, quic_enabled_config);
```

Reply:
yeah that makes sense. no worries, then.